src/serverenvironment.cpp

```diff
@@ -509,9 +509,9 @@ bool ServerEnvironment::line_of_sight(v3f pos1, v3f pos2, float stepsize, bool n
 
 		MapNode n = getMap().getNodeNoEx(pos);
 
-		if (n.drawtype != NDT_AIRLIKE) {
+		if (m_gamedef->ndef()->get(n).drawtype != NDT_AIRLIKE) {
```

Comment:
I am really not sure what the performance impact would be but I suspect it may be significant. This really need to be benchmarked as a number of mods (including my own) rely on the fact the this function is currently extremely fast. IMO a better implementation exists in the ray-cast PR which already has one approval, it just needs a rebase.